### ISSUE TITLE
adding descriptions of the algorithms principles in the doc

### ISSUE THREAD

**Issue by hansen7** [metric-learn/metric-learn#172]
#### Description
Hi, guys. I am doing some research in metrics learning and find your module is super helpful! Would you mind if I add some more descriptions about the theory for each algorithms?
For instance, in the MMC, it "aims at maximising the sum of distances between dissimilar points while keeping the sum of distances between similar examples small"

[A Survey on Metric Learning for Feature Vectors and Structured Data.pdf](https://github.com/metric-learn/metric-learn/files/2876742/A.Survey.on.Metric.Learning.for.Feature.Vectors.and.Structured.Data.pdf)

**Comment by perimosocordiae** [comment]
Sure, feel free to open a Pull Request with your proposed changes. I'm glad you're finding this code useful!

**Comment by bellet** [comment]
Definitely agree; I think we should have an intuitive textual description of what each algorithm tries to do, and even a short mathematical description of the problem for people who want to know more about the exact formulation, as done in sklearn. We should aim at this for the 0.5.0 release
